acceptance/tests/apply/augeas/puppet.rb

```diff
@@ -0,0 +1,40 @@
+test_name "Augeas puppet configuration" do
+
+  tag 'risk:medium'
+
+  confine :except, :platform => 'windows'
+
+  teardown do
+    hosts.each do |host|
+      on host, "cat /tmp/puppet.conf.bak > #{host.puppet['confdir']}/puppet.conf && rm /tmp/puppet.conf.bak"
+    end
+  end
+
+  hosts.each do |host|
+    step "Backup the puppet config"
+    on host, "cp #{host.puppet['confdir']}/puppet.conf /tmp/puppet.conf.bak"
+    #Frictionless installs of an agent leave the agent without a [main] or [agent] section in the puppet.conf file.
+    #This invalid formatted puppet.conf file is fine for most things, but will break when augeas runs agianst it.
+    #This test is for testing augues, and shouldn't be concerend if puppet.conf is perfectly formatted.
+    #So we check to see if there is a main section, and if not we add that and an agent section in.
+    if on(host, "grep '\\[main\\]' #{host.puppet['confdir']}/puppet.conf" , :acceptable_exit_codes => [0, 1]).exit_code == 1
+      #sed option is needed because on OSX using the -i option is included by default, and using it will cause errors.
+      sed_option = "-i"
+      if host['platform'] =~ /osx/
+        sed_option = ""
+      end
+      on(host, "sed #{sed_option} '1s/^/\[main\]\\n/' #{host.puppet['confdir']}/puppet.conf")
+      on(host, "puppet config set runinterval 10 --section agent")
+    end
+
+    step "Modify the puppet.conf file"
+    manifest = <<EOF
+  augeas { 'puppet agent noop mode':
+    context => "/files#{host.puppet['confdir']}/puppet.conf/agent",
+    changes => 'set noop true',
+  }
+EOF
```

Comment:
As per the conversation on PA-71, currently on AIX the {{incl}} and the {{lens}} parameters are needed in order for augeas to load correctly. We should change this manifest to include these.

```
     manifest = <<EOF
   augeas { 'puppet agent noop mode':
     context => "/files#{host.puppet['confdir']}/puppet.conf/agent",
+    incl    => "/etc/puppetlabs/puppet/puppet.conf",
+    lens    => 'Puppet.lns',
     changes => 'set noop true',
   }
 EOF
```